cli/depext.ml

```diff
@@ -31,21 +32,41 @@ let run (`Root root) (`Lockfile explicit_lockfile) dry_run (`Yes yes) () =
             else acc)
           ~init:OpamSysPkg.Set.empty depexts
       in
-      let* pkgs = available_packages pkgs in
-      match OpamSysPkg.Set.elements pkgs with
-      | [] -> Ok ()
-      | pkgs_list ->
-          let pkgs_str = List.map ~f:OpamSysPkg.to_string pkgs_list in
-          if dry_run then (
-            Fmt.pr "%s\n%!" (String.concat ~sep:" " pkgs_str);
-            Ok ())
-          else if should_install ~yes pkgs_str then
-            try
-              OpamCoreConfig.update ~confirm_level:`unsafe_yes ();
-              OpamSysInteract.install pkgs;
-              Ok ()
-            with Failure msg -> Error (`Msg msg)
-          else Ok ())
+      let enabled =
+        match resolve_depexts with
+        | `True -> true
+        | `False -> false
+        | `Config -> OpamFile.Config.depext global_state.config
+      in
+      if enabled then
+        let* pkgs = available_packages pkgs in
+        match OpamSysPkg.Set.elements pkgs with
+        | [] -> Ok ()
+        | pkgs_list ->
+            let pkgs_str = List.map ~f:OpamSysPkg.to_string pkgs_list in
+            if dry_run then (
+              Fmt.pr "%s\n%!" (String.concat ~sep:" " pkgs_str);
+              Ok ())
+            else if should_install ~yes pkgs_str then
+              try
+                OpamCoreConfig.update ~confirm_level:`unsafe_yes ();
+                OpamSysInteract.install pkgs;
+                Ok ()
+              with Failure msg -> Error (`Msg msg)
+            else Ok ()
+      else (
+        Logs.warn (fun l ->
+            let message =
+              "Refusing to check for external dependencies as depext is \
+               disabled.\n\
+               The following packages may need to be installed manually:"
+            in
+            l "%s\n%s" message
+              (String.concat ~sep:"\n"
```

Comment:
You can use `Fmt.list ~sep:(Fmt.any "\n") string` to avoid that concat.

Or even (untested, but I enjoy how much one can do with `Fmt` combinators):
```ocaml
let pp_sys_pkg = Fmt.of_to_string OpamSysPkg.to_string in
let pp_item ppf = Fmt.pf ppf "- %a" pp_sys_pkg in
l "%s\n%a" message (Fmt.list ~sep:(Fmt.any "\n") pp_item) (OpamSysPkg.Set.elements pkgs)
```